### PR TITLE
Fix for thrown items dealing damage twice to first target

### PR DESCRIPTION
**Pull request by Plykiya**
## About the PR

All I did was delete a function call

## Why / Balance

Fixes https://github.com/space-wizards/space-station-14/issues/26022

## Technical details

#18574 called LandComponent() in DamageOtherOnHitSystem as soon as it hits literally anything and regenerated contacts to stop items phasing through walls. Any subsequent hit on other entities will also trigger LandComponent() but because it's already "Landed" from the first call, it doesn't do anything. LandComponent() only triggers on the first hit, regenerates the contacts, and hits the first target twice. Additionally, it doesn't even "land" the item, that's calculated elsewhere.

#18576 fixed what the previous PR was intending to do but did not remove the behavior where it calls LandComponent() on hit because it is seemingly necessary for reasons related to I don't really know. Items no longer phase through walls as a result of this PR, regardless of LandComponent() being called.

https://github.com/space-wizards/space-station-14/pull/20700 later on made it so that it calls LandComponent() in ThrownItemSystem when the time is appropriate in the lifecycle of the throw and not on the first hit. So LandComponent() is successfully being called when flyTime for the item is less than the currentTime in the game.

In summary, we have no need for that LandComponent() in DamageOtherOnHitSystem anymore so we can just delete it. The only thing it accomplished was duplicating damage on the first hit of an entity.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**FOR DEMONSTRATION PURPOSES I'M USING A PIERCING SPEAR**
Before:

https://github.com/user-attachments/assets/ccea3cff-c556-48b5-9844-92cf175bb5a5

After:

https://github.com/user-attachments/assets/da5303ea-55ba-4878-8327-b09778313ada

## Breaking changes

nothing

**Changelog**

:cl:
- fix: You no longer deal double damage to your first target when throwing an item.

### CODE REVIEW

**Comment by Plykiya** [comment]
Yes, I've tested throwing items at walls to make sure they don't do anything weird. Everything behaves normally. Nothing is doing anything weird. I've thrown every single item I can find. You can still throw items into trash cans, things still bounce off walls, throwing things in no grav is still fine, etc etc.

**Comment by Plykiya** [comment]
My feelings are summarized here:
image

**Review by Brandon-Huu — CHANGES_REQUESTED** [review]
https://github.com/space-wizards/space-station-14/pull/30115/files#diff-f11356ab6fc02b074291be6f127ce39af7ebbc673b7cc9cca6f4bcab3cc61b9dL35

```C#
if (TerminatingOrDeleted(args.Target))
    return;
````

**Comment by EmoGarbage404** [comment]
as a wise man once said in another throwing-related PR: i trust you with my life.
